### DISCUSSION TITLE
Laravel csrf token not working on linux 419 page expired but work fine in windows

### DISCUSSION THREAD

**Discussion by bhiantara** [laravel/framework#39826] [Q&A]
- Laravel Version: 6.20.26
- PHP Version: 8.0.13
- Database Driver & Version:  MariaDB

### Description:
My project laravel work fine in windows but when i'm run my project in linux  getting problem with csrf token not work after submit form, for example  after submit login form will be direct to  419 page expired.  So i try to inspect element  and check csrf token already inside in form element, nothing wrong with code.
i'm already try to change permission in folder still not work. 
-chmod -R 777 storage
-chmod -R 777 vendor
-chmod -R 777 bootstrap/cache

**Comment by endo64** [comment]
I have the same issue (Laravel 6.20.44), I literally tried everything with no luck:

php artisan clear, cache:clear, view:clear, optimize, ...
manually deleted all cache files including bootstrap/cache
give full access to all folders
deleted vendor folder and made a fresh one with composer install
checked the token on the page and in the file in session folder, they are same.
checked the request and see token is sent correctly
checked system date/time
cleared browser cache, checked with another browser, in private mode
tried every combination of ENV settings found on internet
tried with Apache server, artisan serve, from local pc, from a remote pc

it only works if I disabled CSRF middleware.
Oh and I see no error log in anywhere.

How is that possible?! 😭😢

**Reply by jeremy0505** [reply]
Hi Endo64, did you manage to resolve the issue? I am facing something very similar (with so many of the steps you outline being performed) and am very frustrated! I'm on Laravel 9x with PHP 8x. Googling extensively on "laravel http code 419" and hoping for some inspiration!

**Reply by endo64** [reply]
@jeremy0505 I'm afraid I did not resolve it :(

**Reply by jeremy0505** [reply]
So did you abandon that project?

My issues were resolved by upgrade from laravel 9 to 10 – however that was just good fortune, I have no idea what was actually "broken" before! Which as every developer knows is...concerning.

cheers
j

**Reply by endo64** [reply]
I can't exactly remember what did I do.. I think it didn't work on my PC but it worked on the server, so continued working on the (prod🤫) server.

Thanks for the note how did you resolve it.
